test/fast/Unit tests/nvm_supports_xz

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+cleanup () {
+  rm ../../xz-test/{which,command,awk,rm}
+  rmdir ../../xz-test
+}
+die () { echo $@ ; exit 1; }
+
+. ../../../nvm.sh
+
+mkdir ../../xz-test
+touch ../../xz-test/xz
+chmod +x ../../xz-test/xz
+ln -s /usr/bin/which ../../xz-test/which
+ln -s /usr/bin/command ../../xz-test/command
+ln -s /usr/bin/awk ../../xz-test/awk
+ln -s $(which rm) ../../xz-test/rm
+
+
+OLDPATH=$PATH
+export PATH=../../xz-test
```

Comment:
Overwriting PATH means that any `nvm` internals relying on things that should be available via PATH will break - doing it in tests makes future refactoring of `nvm` more brittle, and doesn't accurately represent the user's environment.

I think that for the tests where `zx` is *not* installed - which ideally are in a separate test file from the ones testing that it *is* installed - should make whatever assertions or stubs are needed to cover that case.